### PR TITLE
FS-1274: adds locale selection functionality

### PR DESCRIPTION
**Pull request by gidsg**
Adds locale selection functionality for translations, usage in apps is as per README.

(Adapted from the spike done on Frontend https://github.com/communitiesuk/funding-service-design-frontend/compare/welsh)

### CODE REVIEW

**Review by srh-sloan — APPROVED** [review]
Looks good!

File: README.md [hunk 1]
```diff
@@ -152,3 +152,54 @@ Custom checks can be created as subclasses of `CheckerInterface` and should cont
...
+
+    babel = Babel(flask_app)
+    babel.locale_selector_func = get_lang
+    LanguageSelector(flask_app)
+
+The `get_lang()` function reads the user-selected language from a cookie (if set by `LanguageSelector` - see below), or if that is not present uses a built-in function of babel to negotiate the language based on the request headers and the supported languages.
+
+`LanguageSelector` creates an additional route `/language/<language>` that sets the user's selected language in a cookie. Used the cookie rather than the session so it can be shared across the microservices.
+
+Set `SESSION_COOKIE_DOMAIN` on the app to the domain you want to set the cookie on.
```
Comment (harryyo) [inline README.md:172]: So on the authenticator we have a config var called "COOKIE_DOMAIN" which is what sets the dot-prefixed domain for multi-service cookies (like the fsd_auth_token cookie). Presumably this should be the same domain? Shall we conform the name of the key to the same thing (whichever)and maybe even set it to a default in commonconfig.py?

File: README.md [hunk 2]
```diff
@@ -152,3 +152,54 @@ Custom checks can be created as subclasses of `CheckerInterface` and should cont
...
+
+1. To add translations for strings, edit the `messages.po` file.
+
+        #: app/templates/index.html:10
+        msgid "Good Morning"
+        msgstr "Bore da"
+
+    where `msgid` is the english version of the string from the original template file, and `msgstr` is the translation of the string.
+
+1. Once the translations are ready, use `compile` to generate the binary for use at runtime:
```
Comment (Adam-W1) [inline README.md:203]: very clear docs 👏

**Review by harryyo — CHANGES_REQUESTED** [review]
Great, a lot to get your head around, just a couple of questions, wondering about the SESSION_COOKIE_DOMAIN var mainly

File: fsd_utils/config/commonconfig.py [hunk 3]
```diff
@@ -17,7 +17,6 @@ class CommonConfig:
     # ---------------
     SECRET_KEY = os.getenv("SECRET_KEY", "secret_key")
     SESSION_COOKIE_NAME = os.getenv("SESSION_COOKIE_NAME", "session_cookie")
-    FORCE_HTTPS = True
```
[the remark this answers is not in the document]
Comment (gidsg) [reply fsd_utils/config/commonconfig.py:20]: Ah, yes, didn't mean to include this, but it is redundant as PaaS enforces this anyway.

File: README.md (hunk 1 above)
Comment (gidsg) [reply README.md:172]: Don't think we want to default this as it's going to vary per environment, but yeah we probably should use the same variable.

File: README.md [hunk 4]
```diff
@@ -152,3 +152,54 @@ Custom checks can be created as subclasses of `CheckerInterface` and should cont
...
+`LanguageSelector` creates an additional route `/language/<language>` that sets the user's selected language in a cookie. Used the cookie rather than the session so it can be shared across the microservices.
+
+Set `SESSION_COOKIE_DOMAIN` on the app to the domain you want to set the cookie on.
+
+### Creating Translations
+1. Add `trans` tags around items in your jinja html file that you want to translate. What's contained in the `trans` tag should be the english version of this text. eg:
+
+        A greeting is: {% trans %}Good Morning{% endtrans %}
+
+1. Generate a new translations template file (`messages.pot`)
```
[the remark this answers is not in the document]
Comment (gidsg) [reply README.md:179]: This uses the auto-numbering feature of Github Markdown, see the rich preview.

**Pull request merged**